src/util/block_tools.py

```diff
@@ -285,25 +269,7 @@ def get_consecutive_blocks(
         if num_blocks == 0:
             return block_list
 
-        for full_block in block_list:
-            if full_block.height == 0:
-                difficulty = uint64(constants.DIFFICULTY_STARTING)
-            else:
-                difficulty = full_block.weight - block_list[full_block.height - 1].weight
-            quality_str = full_block.reward_chain_sub_block.proof_of_space.verify_and_get_quality_string(constants)
-            required_iters: uint64 = calculate_iterations_quality(
-                quality_str,
-                full_block.reward_chain_sub_block.proof_of_space.size,
-                difficulty,
-            )
-            sub_blocks[full_block.header_hash] = full_block_to_sub_block_record(
-                constants,
-                sub_blocks,
-                height_to_hash,
-                full_block,
-                required_iters,
-            )
-            height_to_hash[uint32(full_block.height)] = full_block.header_hash
+        difficulty = get_difficulty(block_list, constants, height_to_hash, sub_blocks)
```

Comment:
You are passing height to hash and modifying it within the function, that's not a good idea (it's better to have no side effects in small functions like this one)